src/js/draw.js

```diff
@@ -522,3 +499,62 @@ export function draw_metarelation(draw_context, mei_graph, g_elem) {
 
   return added
 }
+
+// Recursively add hover class to elements and their children
+function addHoverClassToChildren(element, isRoot, isPrimary, draw_context, mei_graph) {
+  if (!element) return
+
+  // For meta-relations
+  if (element.classList.contains('metarelation') || element.classList.contains('relation')) {
+    // Add hover class to the meta-relation itself
+    if (!isRoot) element.classList.add(isPrimary ? 'extrarelationhover' : 'relationhover')
+
+    // Get the corresponding MEI node using the oldid attribute
+    let meiNode = get_by_id(mei_graph, element.getAttribute('oldid') || element.id)
+
+    // Recursively add hover class to the children
+    let primaries = relation_primaries(mei_graph, meiNode).map(
+      (e) => document.getElementById(id_in_svg(draw_context, node_to_note_id(e)))
+    )
+    let secondaries = relation_secondaries(mei_graph, meiNode).map(
+      (e) => document.getElementById(id_in_svg(draw_context, node_to_note_id(e)))
+    )
+    console.log(primaries, secondaries)
+
+    primaries.forEach(elem => { addHoverClassToChildren(elem, false, true, draw_context, mei_graph) })
+    secondaries.forEach(elem => { addHoverClassToChildren(elem, false, false, draw_context, mei_graph) })
+
+  } else if (element.classList.contains('note')) {
+    // Add hover class to the note itself (note is never a root)
+    element.classList.add(isPrimary ? 'extrahover' : 'selecthover')
+  }
+}
+
+// Recursively add hover class to elements and their children
```

Comment:
Typo: "Recursively ~~add~~ remove class ~~to~~ from elements and their children."